src/frontends/wayland/src/main.rs

```diff
@@ -221,12 +302,59 @@ impl KimeContext {
                 self.vk
                     .modifiers(mods_depressed, mods_latched, mods_locked, group);
             }
-            KeyEvent::RepeatInfo { .. } => {
-                // TODO: repeat key
+            KeyEvent::RepeatInfo { rate, delay } => {
+                let info = RepeatInfo { rate, delay };
+                let press_state = self.repeat_state.map(|pair| pair.1);
+                self.repeat_state = Some((info, press_state.unwrap_or(PressState::NotPressing)));
             }
             _ => {}
         }
     }
+
+    pub fn handle_timer_ev(&mut self) -> std::io::Result<()> {
+        // Read timer, this MUST be called or timer will be broken
+        self.timer.read()?;
+
+        if let Some((info, ref mut press_state)) = self.repeat_state {
+            let (key, time) = match press_state {
+                PressState::NotPressing => {
+                    log::warn!("Received timer event while pressing no key.");
```

Comment:
Is this warning needed? IMO it could be occur in normal state when Repeat->Timer->KeyPress

Reply:
With "Repeat->Timer->KeyPress", the press_state won't be `NotPressing`. It's not possible AFAIK